lib/Twig/Extension/Core.php

```diff
@@ -652,11 +652,8 @@ function twig_array_merge($arr1, $arr2)
 function twig_slice(Twig_Environment $env, $item, $start, $length = null, $preserveKeys = false)
 {
     if ($item instanceof Traversable) {
-        if ($item instanceof IteratorAggregate) {
+        if ($start >= 0 && $length >= 0 && $item instanceof IteratorAggregate) {
             $item = $item->getIterator();
-        }
-
-        if ($start >= 0 && $length >= 0 && $item instanceof Iterator) {
```

Comment:
I don't think so, testing for `IteratorAggregate` and so change item to an `Iterator` makes the next test `instanceof Iterator` useless, so these can be combined